### PR TITLE
chore: Bump Keycloak dependency

### LINKED ISSUE #15783: Update Keycloak  version to v15
##### **Overview of the feature request**

See https://www.keycloak.org/2021/07/keycloak-1500-released.html

##### **Related issues or PR**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vishal423**
Closes #15783 
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by vishal423** [comment]
There seems an issue with `jhipster-registry` after this upgrade resulting in test failures. I will try to spend sometime later this week to see if we can do something over here.

**Comment by vishal423** [comment]
The latest Keycloak `openid-configuration` endpoint response contains `mtls_endpoint_aliases`, that's not correctly parsed by `oauth2-oidc-sdk` dependency included in the last `JHipster-Registry` release. 

It seems the latest spring boot release `2.5.3` has the correct dependency and that's why it works fine in the new JHipster generated apps. Since `JHipster-Registry` relies on the generator BOM release, which in turn is synchronized with the JHipster release, we have to wait for the next JHipster release. Post that we should bump JHipster Registry to sync the dependencies, test and publish a new release, and then come back over here.

File: generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs [hunk 1]
```diff
@@ -2488,10 +2488,11 @@
     "oauth2DevicePollingInterval": "5",
     "clientSessionIdleTimeout": "0",
     "clientSessionMaxLifespan": "0",
+    "parRequestUriLifespan": "60",
     "clientOfflineSessionIdleTimeout": "0",
     "cibaInterval": "5"
   },
-  "keycloakVersion": "14.0.0",
+  "keycloakVersion": "15.0.0",
```
Comment (DanielFran) [inline generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs:2495]:
```suggestion
  "keycloakVersion": "15.0.2",
```

File: generators/generator-constants.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ const DOCKER_NEO4J = 'neo4j:4.2.7';
 const DOCKER_HAZELCAST_MANAGEMENT_CENTER = 'hazelcast/management-center:4.2021.04';
 const DOCKER_MEMCACHED = 'memcached:1.6.9-alpine';
 const DOCKER_REDIS = 'redis:6.2.4';
-const DOCKER_KEYCLOAK = 'jboss/keycloak:14.0.0'; // The version should match the attribute 'keycloakVersion' from /docker-compose/templates/realm-config/jhipster-realm.json.ejs and /server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs
+const DOCKER_KEYCLOAK = 'jboss/keycloak:15.0.0'; // The version should match the attribute 'keycloakVersion' from /docker-compose/templates/realm-config/jhipster-realm.json.ejs and /server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs
```
Comment (DanielFran) [inline generators/generator-constants.js:63]:
```suggestion
const DOCKER_KEYCLOAK = 'jboss/keycloak:15.0.2'; // The version should match the attribute 'keycloakVersion' from /docker-compose/templates/realm-config/jhipster-realm.json.ejs and /server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs
```

File: generators/server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs [hunk 3]
```diff
@@ -2485,10 +2485,11 @@
     "oauth2DevicePollingInterval": "5",
     "clientSessionIdleTimeout": "0",
     "clientSessionMaxLifespan": "0",
+    "parRequestUriLifespan": "60",
     "clientOfflineSessionIdleTimeout": "0",
     "cibaInterval": "5"
   },
-  "keycloakVersion": "14.0.0",
+  "keycloakVersion": "15.0.0",
```
Comment (DanielFran) [inline generators/server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs:2492]:
```suggestion
  "keycloakVersion": "15.0.2",
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
Ok I thought it was merged.
So, let's do it only for the next release 7.3.0 (instead of 7.2.0)

**Comment by danieljkemp** [comment]
> Ok I thought it was merged.
> So, let's do it only for the next release 7.3.0 (instead of 7.2.0)

Is this still happening in 7.3.0?

**Comment by DanielFran** [comment]
@vishal423 Can you finalize the Dev and sync with registry?

**Comment by vishal423** [comment]
> 
> 
> @vishal423 Can you finalize the Dev and sync with registry?

@DanielFran, we have to follow below steps:
- Check-in fix on registry ( there is a PR that unblocks this)
- Release registry
- Update registry dependency over here and re-run the PR tests.

**Comment by mshima** [comment]
@vishal423 there is no need to release.
With https://github.com/jhipster/jhipster-registry/pull/514, we can use snapshot docker image.

**Comment by vishal423** [comment]
I see all tests passing (ignoring npm snapshot failures as they are due to temporary change to point to jhipster-registry snapshot). @pascalgrimaud , once you release the next jhipster-registry version, and update in generator, we can merge this PR.

I will revert temporary snapshot changes from this branch

**Comment by vishal423** [comment]
> 
> 
> I see all tests passing (ignoring npm snapshot failures as they are due to temporary change to point to jhipster-registry snapshot). @pascalgrimaud , once you release the next jhipster-registry version, and update in generator, we can merge this PR.
> 
> I will revert temporary snapshot changes from this branch

Or if we want to have snapshot integrated in generator and later change to released version, then, I can update jest snapshot tests to pass so that we can merge this PR as is

**Comment by mshima** [comment]
+1 to use the snapshot.
I will try to sync jhipster-registry with main and improve sync process for next releases.

**Review by DanielFran — APPROVED** [review]

**Comment by vishal423** [comment]
Are we good over here?

**Comment by vishal423** [comment]
@pascalgrimaud bounty claimed @ https://opencollective.com/generator-jhipster/expenses/61026

**Comment by pascalgrimaud** [comment]
@vishal423 : approved

**Pull request merged**